lazyflow/classifiers/vigraRfAdaptiveMaskPixelwiseClassifier.py

```diff
@@ -0,0 +1,198 @@
+        logger.debug( 'training pixel-wise vigra RF' )
+        
+        all_features = numpy.ndarray( shape=(0, feature_images[0].shape[-1]), dtype=numpy.float32 )
+        all_labels = numpy.ndarray( shape=(0,1), dtype=numpy.uint32 )
+
+        # Extract label points and corresponding feature vectors
+        for feature_image, label_image in zip(feature_images, label_images):
+            label_coords = numpy.nonzero(label_image)[:-1] # discard channel
+            label_vector = label_image[label_coords]
+            feature_matrix = feature_image[label_coords]
+            all_features = numpy.concatenate((all_features, feature_matrix), axis=0)
+            all_labels = numpy.concatenate((all_labels, label_vector), axis=0)
+        
+        # Save for future reference
+        known_labels = numpy.unique(all_labels)
+
+        assert len(all_features) == len(all_labels)
+        classifier = vigra.learning.RandomForest(*self._args, **self._kwargs)
+        classifier.learnRF(all_features, all_labels)
+
+        return VigraRfAdaptiveMaskPixelwiseClassifier( classifier, known_labels )
+
+    def get_halo_shape(self, data_axes):
+        # No halo necessary, but since this classifier is for testing purposes, let's add one anyway.
+        halo = tuple(range( len(data_axes)-1 ))
+        return halo + (0,) # (no channel halo)
+
+    def estimated_ram_usage_per_requested_predictionchannel(self):
+        return 4
+
+    @property
+    def description(self):
+        temp_rf = vigra.learning.RandomForest( *self._args, **self._kwargs )
+        return "Vigra Random Forest ({} trees)".format( temp_rf.treeCount() )
+
+    def __eq__(self, other):
+        return (    isinstance(other, type(self))
+                and self._args == other._args
+                and self._kwargs == other._kwargs )
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+assert issubclass( VigraRfAdaptiveMaskPixelwiseClassifierFactory, LazyflowPixelwiseClassifierFactoryABC )
+
+class VigraRfAdaptiveMaskPixelwiseClassifier(LazyflowPixelwiseClassifierABC):    
+    """
+    Adapt the vigra RandomForest class to the interface lazyflow expects.
+    """
+    def __init__(self, vigra_rf, known_labels):
+        self._known_labels = known_labels
+        self._vigra_rf = vigra_rf
+    
+    def predict_probabilities_pixelwise(self, X, axistags=None): 
+        logger.debug( 'predicting PIXELWISE vigra RF' )
+        
+        FRAME_SPAN = 10 
+        DILATION_RADIUS = 50
+        BACKGROUND_LABEL = 1  
+        
+        # Allocate memory for probability volume and mask    
+        prob_vol = numpy.zeros((X.shape[:-1]+(len(self._known_labels),) ))           
+        mask = numpy.ones(numpy.prod(X.shape[1:-1]))
+        
+        frm_cnt = 0
+          
+        for X_t in X:    
+            if frm_cnt % FRAME_SPAN == 0 :
+                mask = numpy.ones(numpy.prod(X.shape[1:-1]))
```

Comment:
When using `numpy.zeros()` or `numpy.ones()`, always specify an explicit dtype.  The default dtype is `float64`, which is larger than necessary here, and therefore slower.  Need to fix this line and a few others, below.